### PR TITLE
[Feature] Add with cp to mit and vit

### PR DESCRIPTION
**Pull request by jiangyitong**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support with cp for mit and vit backbone
## Modification
add with_cp parameter and modify the transformerencoder 
## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmseg/models/backbones/mit.py [hunk 1]
```diff
@@ -275,9 +279,19 @@ def __init__(self,
...
     def forward(self, x, hw_shape):
-        x = self.attn(self.norm1(x), hw_shape, identity=x)
-        x = self.ffn(self.norm2(x), hw_shape, identity=x)
+
+        def _inner_forward(x):
+            x = self.attn(self.norm1(x), hw_shape, identity=x)
+            x = self.ffn(self.norm2(x), hw_shape, identity=x)
+            return x
+
+        if self.with_cp and x.requires_grad:
```
Comment (MengzhangLI) [inline mmseg/models/backbones/mit.py:291]: Please add unit test. Thanks in advance!

File: mmseg/models/backbones/mit.py [hunk 2]
```diff
@@ -275,9 +279,19 @@ def __init__(self,
...
+            x = cp.checkpoint(_inner_forward, x)
```
Comment (MengzhangLI) [inline mmseg/models/backbones/mit.py:292]: Please add unit test. Thanks in advance!

**Comment by Junjun2016** [comment]
@jiangyitong 
Please sign the CLA.

**Review by Junjun2016 — APPROVED** [review]

**Comment by MengzhangLI** [comment]
Look Good To Me. 

I would make **channel number**, **width** and **height** of input and weight of model smaller in unit test to pass CI more quickly in the other pr.

**Review by MengzhangLI — APPROVED** [review]

**Pull request merged**